### DISCUSSION TITLE
Does a BG collate all responses from BPP and send it to BAP?

### DISCUSSION THREAD

**Discussion by aks30** [beckn/DSEP-Specification#36] [Q&A]
When a BAP fires a search query to BG, a BG then broadcasts the same to all BPPs registered with the same domain as specified by a BAP. Once the BPP is ready with the response or search results does that go to BAP directly from BPP or it goes to a BG which then forwards the same to BAP?

In case the BPP returns a response to BG, does a BG wait for a response from other BPPs, collate them and then send it to BAP? In case it does not collate what is the use of the BPP response being forwarded to BG, why cannot a BPP directly send the search results to BAP? 

Cc - @vijiurs  @kiranharidas187

**Comment by ravi-prakash-v** [comment]
No. The BG simply forwards the responses received from the BPP to the BAP.

**Reply by aks30** [reply]
@core-wg-admin  In case it does not collate what is the use of the BPP response being forwarded to BG, why cannot a BPP directly send the search results to BAP?
